lulesh-v1.0/RAJA/CMakeLists.txt

```diff
@@ -23,6 +23,8 @@ if (ENABLE_CUDA)
   set (lulesh_depends
     ${lulesh_depends}
     cuda)
+  set (lulesh_compile_options
```

Comment:
@rhornung67 @mdavis36 I think it might be better to set these options on the `RAJA` library target (in RAJA).

Reply:
I agree.

Reply:
Okay so I gave this a go and it works so long as the target also depends on cuda and therefore builds cuda objects. However whenever ENABLE_CUDA=On, the target depending on raja does not depend on cuda, and is therefore building C++ objects, it will fail (i.e. baseline lulesh targets fail to compile because they don't depend on cuda and therefore don't recognize the compiler options). 

I’m not sure if there is a way to set target_compile_options only when the interfacing target is going to be a cuda object? We could force targets to depend on cuda when they depend on RAJA and ENABLE_CUDA=On, but I'm not sure if that's what we want either. What do you guys think?